mini-python-ocaml/compile.ml

```diff
@@ -218,32 +292,32 @@ let rec compile_expr (env: env_t) (expr: Ast.texpr) : X86_64.text * X86_64.data
     end
   | TEunop (op, e) ->
     begin match op with
-      | Uneg ->
-        let text_code, data_code, expr_type = compile_expr env e in
-        begin match expr_type with
-        | `int ->
-          text_code ++
-          movq (ind rax) (reg rdi) ++
-          negq (reg rdi) ++
-          movq (imm byte) (reg rsi) ++
-          call "malloc_wrapper",
-          data_code, `int
-        | _ ->
-          failwith "Unsupported Uneg"
-        end
-      | Unot ->
-        let text_code, data_code, expr_type = compile_expr env e in
-        begin match expr_type with
-        | `bool ->
-          text_code ++
-          movq (ind rax) (reg rdi) ++
-          xorq (imm 1) (reg rdi) ++
-          movq (imm byte) (reg rsi) ++
-          call "malloc_wrapper",
-          data_code, `bool
-        | _ ->
-          failwith "Unsupported Unot"
-        end
+    | Uneg ->
+      let text_code, data_code, expr_type = compile_expr env e in
+      begin match expr_type with
+      | `int ->
+        text_code ++
+        movq (ind rax) (reg rdi) ++
+        negq (reg rdi) ++
+        movq (imm byte) (reg rsi) ++
+        call "malloc_wrapper",
+        data_code, `int
+      | _ ->
+        failwith "Unsupported Uneg"
+      end
+    | Unot ->
+      let text_code, data_code, expr_type = compile_expr env e in
+      begin match expr_type with
+      | `bool ->
+        text_code ++
+        movq (ind rax) (reg rdi) ++
+        xorq (imm 1) (reg rdi) ++
+        movq (imm byte) (reg rsi) ++
+        call "malloc_wrapper",
+        data_code, `bool
+      | _ ->
+        failwith "Unsupported Unot"
+      end
```

Comment:
`Uneg`, `Unot` need to operate on the value, not on the tag.
Currently, they operate on the tag.